### ISSUE TITLE
Not working with Tesla C2050

### ISSUE THREAD

**Issue by ghost** [cbuchner1/ccminer#6]
Hi,
Thanks for the work on X11. I am trying to mine Darkcoin with Tesla C2050, Win 7 x64. It seems to me that the newest release 1.0 is using CPUs rather than GPUs. I did see some hash rate (very high), but no shares accepted in the pool.

**Comment by KBomba** [comment]
The Tesla C2050 is Compute Capability 2.0, and that has been dropped for latest CCMiner releases IIRC.

**Comment by ghost** [comment]
Thank you, just solved it by compiling with Compute 2.0.

**Comment by cbuchner1** [comment]
a lot of algorithms in ccMiner wouldn't support Compute 2.0 at all (some
rewrite/redesign needed)

2014-05-11 16:45 GMT+02:00 whuplus notifications@github.com:

**Comment by KBomba** [comment]
Interesting. How did you compile it exactly? I tried running it at compute_20,sm_20, tried changing the MAXWELL_OR_FERMI constant, but it keeps erroring out with a message about Cuda Toolkit 5.5 not liking it. Will I need an older toolkit?
BTW, just out curiosity, what's your hashrate at x11? ;)

**Comment by ghost** [comment]
Not at all. I just dropped support for Jackpotcoin, Animecoin and Quarkcoin, and deleted two 'compactionTest.cu' files.

Currently 1.3MH/s with the x64 release build (compute_20,sm_20).

**Comment by KBomba** [comment]
Yessireh! Thx, that did it for me ;)

**Comment by cdp1337** [comment]
whuplus, where did you disable those algorithms?  I can't find the options for configure anywhere, in fact I can't even find any build documentation at all!

**Comment by ghost** [comment]
Actually, I rewrote "cpu-miner.c" line 866-918 and removed some sources files that cause compiling errors.

**Comment by cdp1337** [comment]
Oh yeah?  Would you possibly be so kind as to publish a fork of this? (or at very least a patch file :) )

**Comment by ghost** [comment]
Sure, but I am not familiar with Github and it's very late. I'll try it tomorrow. ^_^

**Comment by KBomba** [comment]
This is what I did for compute 2.x @ my fork: https://github.com/KBomba/ccminer-KBomba/commit/f91269c1d02e46e3172b30bfc1bf7474b03345d6
Many thanks for whuplus!

**Comment by cdp1337** [comment]
Thanks, compiling now!  I had to rip out a few things from the Makefile, seeing as you're using an IDE to compile the code, only understandable.

**Comment by x46085** [comment]
Hey fellas, I'm running into same issue with the Makefile since I'm trying to compile for Ubuntu, I get the error below. cdp1337, can you explain what you ripped out to help it compile? I'm targeting x11 on Nvidia 560Ti's as my end goal. Thanks so much! 

make[2]: **\* No rule to make target `JHA/cuda_jha_compactionTest.cu', needed by`JHA/cuda_jha_compactionTest.o'.  Stop.

**Comment by x46085** [comment]
p.s. I'm happy to throw a donation at whoever can help me compile this on Ubuntu 13.10, I'm sure it's an easy fix but I don't have time to learn all the make file and figure it out. $100 in BTC sound fair?

**Comment by x46085** [comment]
More info on this, I need it to compile for Computer Capability 2.1 since I'm using Nvidia 560ti's. I was trying to build KBomba's release: https://github.com/KBomba/ccminer-KBomba/releases/tag/v1.0.2x initially, but whatever will work best for my needs I'm totally open to options. Thoughts?

**Comment by cbuchner1** [comment]
When compiling on Linux, the key is to modify Makefile.am.
Alter the places where it says compute_35, sm_35 and compute_30, sm_30 into
compute_21, sm_21

remove the all compactionTest.cu modules from Makefile.am

comment out the code lines where the (now missing) functions in
compactionTest are called.

The Quark, Anime, Jackpotcoin algorithms will not work afterwards.

X11 and others remain functional.

2014-06-05 18:00 GMT+02:00 Daniel Driscoll notifications@github.com:

**Comment by x46085** [comment]
Awesome thanks I'll check it out.

**Comment by cdp1337** [comment]
Yeah, what they said.  If I get time soon I'll put up a fork of my changes, but my rig that handled the nvidia compiling jobs got commandeered for use in the NAS, so instead I'm stuck with an ancient celeron processor for testing with :/

**Comment by x46085** [comment]
Well, I think you lost me at the "comment out code lines" part, because after adjusting as you described I get this, and can't seem to find the debug line that prints it:

gcc -std=gnu99 -DHAVE_CONFIG_H -I.  -msse2  -fopenmp -pthread -fno-strict-aliasing -I./compat/jansson -DSCRYPT_KECCAK512 -DSCRYPT_CHACHA -DSCRYPT_CHOOSE_COMPILETIME   -g -O2 -MT ccminer-sha2.o -MD -MP -MF .deps/ccminer-sha2.Tpo -c -o ccminer-sha2.o `test -f 'sha2.c' || echo './'`sha2.c
mv -f .deps/ccminer-sha2.Tpo .deps/ccminer-sha2.Po
nvcc -g -O2 -I . -Xptxas "-abi=no -v" -gencode=arch=compute_35,code=\"sm_35,compute_35\" -gencode=arch=compute_21,code=\"sm_21,compute_21\" --maxrregcount=80 --ptxas-options=-v -I./compat/jansson -o heavy/heavy.o -c heavy/heavy.cu
nvcc fatal   : Unsupported gpu architecture 'compute_21'
make[2]: **\* [heavy/heavy.o] Error 255
make[2]: Leaving directory `/home/lemma/cc102'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory`/home/lemma/cc102'
make: **\* [all] Error 2

**Comment by cbuchner1** [comment]
hmm it would appear compute_21 is not a valid compute specification, use
compute_20 and sm_21

Christian

2014-06-06 6:38 GMT+02:00 Daniel Driscoll notifications@github.com:

**Comment by x46085** [comment]
Awesome, that did the trick. Now I've just gotten figure out where to comment out these function calls, will let you know when I do.

**Comment by x46085** [comment]
Finally figured it out, removing the compilation includes for quark, JHA, etc threw me for a bit. Thanks for the quick responses Christian, donation coming your way.

**Comment by cbravo135** [comment]
I am having a hard time figure out how to modify Makefile.am to compile for Compute 2.0. I am still not sure which includes I am supposed to take out, maybe I am completely misunderstanding what I need to do. I keep getting "error: identifier "__shfl" is undefined"

**Comment by cbuchner1** [comment]
__shfl requires Compute 3.0

here's an older fork that removes the algorithms requiring Compute 3.0

https://github.com/KBomba/ccminer-KBomba

Christian

2014-09-04 2:24 GMT+02:00 Cameron Bravo notifications@github.com:

**Comment by nreal2** [comment]
Has it moved (__shfl requires Compute 3.0 here's an older fork that removes the algorithms requiring Compute 3.0 https://github.com/KBomba/ccminer-KBomba Christian)  ??

Can anyone help meplease to get sources to be able to compile to compute_20 and sm_21, where can i find sources or bin for x86_64 linux ?

**Comment by nreal2** [comment]
Or can someone post that Makefile.am that does the trick please?

**Comment by cbravo135** [comment]
Yeah, I do have the older fork already, I just can not figure out how to
modify the Makefile.am to get it to compile for a Compute 2.0 device.

**Comment by nreal2** [comment]
I could try that, but I suspect it doesnt compile I spend whole day trying
but no luck, have those "older" cards here ...and no windowses never

2014-09-22 20:46 GMT+03:00 Cameron Bravo notifications@github.com:

> Yeah, I do have the older fork already, I just can not figure out how to
> modify the Makefile.am to get it to compile for a Compute 2.0 device.
>
